### PR TITLE
Cypress component tests

### PR DESCRIPTION
**Pull request by freesbi**
### Description

Add Cypress component configuration and tests

### Which Component is affected or generated?

Tests added to the next components:

- danger-button.component
- link-button.component
- icon-button.component
- mad-button.directive
- mad-button-group
- outline-button.component
- primary-button.component
- card.component
- data-table.component (except modal)
- numeric-field.directive
- quick-list.component
- quick-list-compact.component
- reactive-form-quick-list.component
- reactive-form-quick-list-compact.component
- readonly-form-field.component
- readonly-form-field-wrapper.component
- material-action-button.component
- stepper.component
- throttle-click.directive
- toolbar.component

### CODE REVIEW

File: projects/material-addons/src/lib/button/flat-button/link-button.component.cy.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { ButtonModule } from '../button.module';
+import { MatButtonModule } from '@angular/material/button';
+import { Component } from '@angular/core';
+
+const checkButtonBasicAttributes = () => {
```
Comment (xkilpoi) [inline projects/material-addons/src/lib/button/flat-button/link-button.component.cy.ts:5]: I think this is almost a duplicate from above? Maybe we can have only one of them taking some arguments to be configured?

File: projects/material-addons/src/lib/button/icon-button/icon-button.component.cy.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { ButtonModule } from '../button.module';
+import { MatButtonModule } from '@angular/material/button';
+import { MatIconModule } from '@angular/material/icon';
+import { Component } from '@angular/core';
+
+const checkButtonBasicAttributes = () => {
```
Comment (xkilpoi) [inline projects/material-addons/src/lib/button/icon-button/icon-button.component.cy.ts:6]: Also very similar to the other ones :)

File: projects/material-addons/src/lib/button/outline-button/outline-button.component.cy.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { ButtonModule } from '../button.module';
+import { MatButtonModule } from '@angular/material/button';
+import { Component } from '@angular/core';
+import { ThemePalette } from '@angular/material/core';
+
+const checkButtonBasicAttributes = () => {
```
Comment (xkilpoi) [inline projects/material-addons/src/lib/button/outline-button/outline-button.component.cy.ts:6]: Very similar again :)

File: projects/material-addons/src/lib/button/primary-button/primary-button.component.cy.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import { ButtonModule } from '../button.module';
+import { MatButtonModule } from '@angular/material/button';
+import { Component } from '@angular/core';
+
+const checkButtonBasicAttributes = () => {
```
Comment (xkilpoi) [inline projects/material-addons/src/lib/button/primary-button/primary-button.component.cy.ts:5]: Reminder

File: projects/material-addons/src/lib/quick-list/reactive-form-quick-list-compact/reactive-form-quick-list-compact.component.cy.ts [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    }).then((response) => {
+      cy.spy(response.component, 'onItemAdded').as('onItemAddedSpy');
+    });
+    cy.getByCySel('add-first-item-button').click();
+    cy.get('@onItemAddedSpy').should('be.calledOnce');
+    cy.getByCySel('reactive-form-quick-list-item').should('have.length', 1);
+    cy.getByCySel('add-first-item-button').should('not.exist');
+  });
+
+  it('should show delete and add icons on item, but not add button with label', () => {
```
Comment (xkilpoi) [inline projects/material-addons/src/lib/quick-list/reactive-form-quick-list-compact/reactive-form-quick-list-compact.component.cy.ts:174]: Aren't there some tests quite similar to the other quick-list-compact test?

File: projects/material-addons/src/lib/quick-list/reactive-form-quick-list/reactive-form-quick-list.component.cy.ts [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    });
+    cy.getByCySel('add-item-button').should('contain.text', ' Test Label\n');
+  });
+
+  it('should display the correct number of items', () => {
+    mountWrapperComponent();
+    cy.getByCySel('reactive-form-quick-list-item').should('have.length', 2);
+  });
+
+  it('should add an item when add button is clicked', () => {
```
Comment (xkilpoi) [inline projects/material-addons/src/lib/quick-list/reactive-form-quick-list/reactive-form-quick-list.component.cy.ts:167]: Aren't there very similar things inside the quick-list component tests?

File: projects/material-addons/src/lib/stepper/step-header/step-header.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { StepHeaderComponent } from './step-header.component';
+import { FocusMonitor, FocusOrigin } from '@angular/cdk/a11y';
+import { DebugElement } from '@angular/core';
+import { MatIconModule } from '@angular/material/icon';
+import { STEP_STATE } from '@angular/cdk/stepper';
+import { By } from '@angular/platform-browser';
+
+describe('StepHeaderComponent', () => {
```
Comment (xkilpoi) [inline projects/material-addons/src/lib/stepper/step-header/step-header.component.spec.ts:9]: Why don't we use Cypress here?

File: projects/material-addons/src/lib/button/flat-button/link-button.component.cy.ts (hunk 1 above)
Comment (freesbi) [reply projects/material-addons/src/lib/button/flat-button/link-button.component.cy.ts:5]: fixed with button.component.cy.ts tests

File: projects/material-addons/src/lib/button/icon-button/icon-button.component.cy.ts (hunk 2 above)
Comment (freesbi) [reply projects/material-addons/src/lib/button/icon-button/icon-button.component.cy.ts:6]: fixed with button.component.cy.ts tests

File: projects/material-addons/src/lib/button/outline-button/outline-button.component.cy.ts (hunk 3 above)
Comment (freesbi) [reply projects/material-addons/src/lib/button/outline-button/outline-button.component.cy.ts:6]: fixed with button.component.cy.ts tests

File: projects/material-addons/src/lib/button/primary-button/primary-button.component.cy.ts (hunk 4 above)
Comment (freesbi) [reply projects/material-addons/src/lib/button/primary-button/primary-button.component.cy.ts:5]: fixed with button.component.cy.ts tests

File: projects/material-addons/src/lib/stepper/step-header/step-header.component.spec.ts (hunk 7 above)
Comment (freesbi) [reply projects/material-addons/src/lib/stepper/step-header/step-header.component.spec.ts:9]: we are using Cypress in stepper.component.cy.ts. It contains tests for header

File: projects/material-addons/src/lib/quick-list/reactive-form-quick-list/reactive-form-quick-list.component.cy.ts (hunk 6 above)
Comment (freesbi) [reply projects/material-addons/src/lib/quick-list/reactive-form-quick-list/reactive-form-quick-list.component.cy.ts]: Fixed with stepper.component.cy.ts tests

**Comment by freesbi** [comment]
@xkilpoi 
All comments fixed. Ready for review

**Review by xkilpoi — APPROVED** [review]

**Review by Wagei — APPROVED** [review]

**Pull request merged**